### PR TITLE
fix(sql): fix error when using sample by with fill(none)

### PR DESCRIPTION
**Pull request by nwoolmer**
This errors on demo 8.1.0:

```sql
SELECT last(price) value, symbol, timestamp 
FROM trades
WHERE timestamp >= '2024-08-11T10:13:00' 
AND timestamp < '2024-08-11T10:16:00' 
AND (symbol LIKE ('BTC-USD')) 
SAMPLE BY 1m FILL(NONE) ALIGN TO CALENDAR
```

However, if you remove `FILL(NONE)`, it does not.

A check to prevent compilation of an unsupported fill with keyed group by was inadequate.

### CODE REVIEW

**Review by puzpuzpuz — APPROVED** [review]

File: core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java [hunk 1]
```diff
@@ -5564,7 +5564,7 @@ private void guardAgainstFillWithKeyedGroupBy(QueryModel model, ArrayColumnTypes
             return;
         }
 
-        if (curr.getFillValues().size() == 1 && Chars.equals(curr.getFillValues().getQuick(0).token, "none")) {
+        if (curr.getFillValues().size() == 1 && SqlKeywords.isNoneKeyword(curr.getFillValues().getQuick(0).token)) {
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:5567]: nit: there is also `Chars.equals(ast.token, "timestamp_floor")` check in this class. Would be nice to make it case-insensitive too.

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 1 / 1 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|1|1|100.00%|

File: core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java [hunk 2]
```diff
@@ -1762,7 +1762,7 @@ private RecordCursorFactory generateFill(QueryModel model, RecordCursorFactory g
             for (int i = 0, n = model.getBottomUpColumns().size(); i < n; i++) {
                 final QueryColumn col = model.getColumns().getQuick(i);
                 final ExpressionNode ast = col.getAst();
-                if (Chars.equals(ast.token, "timestamp_floor")) {
+                if (isTimestampFloorKeyword(ast.token)) {
```
Comment (puzpuzpuz) [inline core/src/main/java/io/questdb/griffin/SqlCodeGenerator.java:1765]: nit: a `Chars#equalsIgnoreCase()` call would be enough here. A 15 chars-long function matcher isn't very readable.

**Review by puzpuzpuz — DISMISSED** [review]

**Review by puzpuzpuz — DISMISSED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|

**Comment by ideoma** [comment]
@nwoolmer can you have a look into the failing test please?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 2 / 2 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/griffin/SqlCodeGenerator.java|2|2|100.00%|

**Review by ideoma — APPROVED** [review]

**Pull request merged**